api_core/google/api_core/grpc_helpers.py

```diff
@@ -65,6 +65,20 @@ class _StreamingResponseIterator(grpc.Call):
     def __init__(self, wrapped):
         self._wrapped = wrapped
 
+        # This iterator is used in a retry context, and returned outside after init.
+        # gRPC will not throw an exception until the stream is consumed, so we need 
+        # to retrieve the first result, in order to fail, in order to trigger a retry.
+        try:
+            self._stored_first_result = six.next(self._wrapped)
+        except TypeError:
```

Comment:
@tseaver Under test we have a non-iterable. While it is odd that you would call this without it does seem possible? So added a guard here